packages/order-utils/src/market_utils.ts

```diff
@@ -0,0 +1,109 @@
+     * slippageBufferAmount. Iterates from first feeOrder to last. Sort the feeOrders by ascending rate in order to get the subset of
+     * feeOrders that will cost the least ETH.
+     * @param   signedOrders         An array of objects that conform to the SignedOrder interface. All orders should specify ZRX as
+     *                               the makerAsset and WETH as the takerAsset.
+     * @param   orderStates          An array of objects corresponding to the signedOrders parameter that each contain on-chain state
+     *                               relevant to that order.
+     * @param   signedFeeOrders      An array of objects that conform to the SignedOrder interface. All orders should specify ZRX as
+     *                               the makerAsset and WETH as the takerAsset.
+     * @param   feeOrderStates       An array of objects corresponding to the signedOrders parameter that each contain on-chain state
+     *                               relevant to that order.
+     * @param   makerAssetFillAmount The amount of makerAsset desired to be filled.
+     * @param   slippageBufferAmount An additional amount makerAsset to be covered by the result in case of trade collisions or partial fills.
+     * @return  Resulting orders and remaining fill amount that could not be covered by the input.
+     */
+    findFeeOrdersThatCoverFeesForTargetOrders(
+        signedOrders: SignedOrder[],
+        orderStates: OrderRelevantState[],
+        signedFeeOrders: SignedOrder[],
+        feeOrderStates: OrderRelevantState[],
+        slippageBufferAmount: BigNumber = constants.ZERO_AMOUNT,
+    ): { resultOrders: SignedOrder[]; remainingFillAmount: BigNumber } {
+        // type assertions
+        assert.doesConformToSchema('signedOrders', signedOrders, schemas.signedOrdersSchema);
+        assert.doesConformToSchema('signedFeeOrders', signedFeeOrders, schemas.signedOrdersSchema);
+        assert.isBigNumber('slippageBufferAmount', slippageBufferAmount);
+        // calculate total amount of ZRX needed to fill signedOrders
+        const totalFeeAmount = _.reduce(
+            signedOrders,
+            (accFees, order, index) => {
+                const orderState = orderStates[index];
+                const makerAssetAmountAvailable = getMakerAssetAmountAvailable(orderState);
+                const feeToFillMakerAssetAmountAvailable = makerAssetAmountAvailable
+                    .div(order.makerAssetAmount)
+                    .mul(order.takerFee);
+                return accFees.plus(feeToFillMakerAssetAmountAvailable);
+            },
+            constants.ZERO_AMOUNT,
+        );
+        return marketUtils.findOrdersThatCoverMakerAssetFillAmount(
+            signedFeeOrders,
+            feeOrderStates,
+            totalFeeAmount,
+            slippageBufferAmount,
+        );
+    },
+};
+
+const getMakerAssetAmountAvailable = (orderState: OrderRelevantState) => {
```

Comment:
There is a case missing here where the maker does not have enough fees / fee allowance to cover the makerFees

Reply:
@fabioberger @LogvinovLeon does `getOpenOrderRelevantStateAsync` in `OrderStateUtils` already  cover all these edge cases when calculating `remainingFillableMakerAssetAmount` it looks like it might, just want to double check